frontend/services/entities/hosts.ts

```diff
@@ -441,7 +441,7 @@ export default {
     return sendRequest("POST", HOSTS_TRANSFER_BY_FILTER, {
       team_id: teamId,
       filters: {
-        query,
+        query: query || undefined, // Critical: Prevents empty string passed to API
```

Comment:
<<Thread resolved in our meeting>>

Reply:
Summary from @sharon-fdm in slack:
```
Summary of what we agreed and is going as a fix for the Builk Ops critical bug.
1 - The only way to select all hosts for bulk operation of DELETE / TRANSFER is to send an empty filer like this:
{
    "filter": {}
}
This will select all hosts
2 - Only the 4 supported filters are allowed inside the filter field.
These will return an error:
{
    "filter": {
        "foo": 1,
    },
}
ERROR
This is what used to be sent from the UI which selected all hosts, and will not be sent anymore. 
If sent manually by API, the backend will treat it as an error.
{
    "filter": {
        "query": "",
        "status": ""
    }
}
ERROR

```